theano/compile/mode.py

```diff
@@ -159,6 +159,16 @@ def add_requirements(self, fgraph):
         fgraph.attach_feature(gof.DestroyHandler())
 
 
+class AddNoOutputFromInplace(gof.Optimizer):
+    """This optimizer adds to the fgraph a feature that will prevent outputs
+    of a fgraph to be created by performing inplace operations on intermediary
+    variables.
```

Comment:
Tell why this is useful: when the output is preallocated, currently use by scan.